nestjs-counter/apps/worker/src/temporal/workflows.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  continueAsNew,
+  proxyActivities,
+  setHandler,
+  sleep,
+} from '@temporalio/workflow';
+import { ActivitiesService } from '../activities/activities.service';
+// Can't use `@app/shared` here because for some reason Temporal's Webpack
+// build complains that "node_modules/@app/shared" doesn't exist in Jest.
+import {
+  getExchangeRatesQuery,
+  ExchangeRates,
+} from '../../../../libs/shared/src';
+
+const { getExchangeRates } = proxyActivities<ActivitiesService>({
+  startToCloseTimeout: '1 minute',
+});
+
+const maxIterations = 10000;
+
+export async function exchangeRatesWorkflow(
```

Comment:
We're not bringing anything interesting from NextJS to workflows, should we try to use some of their module / provider system to construct workflows?

Reply:
I'm not sure how well using modules and providers in Workflows would work, given that we transpile and bundle Workflows. What if some part of NestJS uses, or starts using, some Node built-ins or something else that is non-deterministic? Plus, there isn't much we can inject, other than maybe `setHandler()`.

Reply:
Having 0 xp with nestjs I figure we can leave this open for now.